### DISCUSSION TITLE
Fluxor / Redux Pattern

### DISCUSSION THREAD

**Discussion by vzdesic** [oqtane/oqtane.framework#912] [Q&A] [answered]
Is there any limitation which blocks me from using Redux Pattern with Oqtane on the site level? My plan is to use Fluxor.

Thanks,
Vedran

**Comment by sbwalker** [comment, marked as answer]
I am not aware of anyone who has tried it previously, so I am not sure if there may be a limitation or not - you will need to give it a try. I think the bigger question is why you feel you need to use Fluxor? What type of state do you need to manage and why is the native state management capabilities in Oqtane insufficient? I have to ask this question because developers often choose tools out of preference rather than necessity - and choosing tools by preference often creates additional challenges for them than if they had familiarized themselves with the native options. Its like a construction project which uses screws, and someone shows up and says they want to use a hammer... sure they might be able to hammer in a screw, but it most likely will not result in an optimal outcome. In general, the loosely coupled architecture of Oqtane tries to limit the restrictions so that people are free to use their preferences, but some approaches will always have better support than others.

**Reply by vzdesic** [reply]
Yes, you are right! Probably native state management capabilities in Oqtane is enough. In my system, I will have many microservices below my UI. I am just investigating Oqtane and possibilities. Thanks a lot for the answer!

**Reply by vzdesic** [reply]
Sorry @sbwalker, I need clarification here. What do you mean by - native state management capabilities in Oqtane. Is it the same like Blazor state management capabilities or something else? Is state saved to Oqtane database? Where I can read about that? I have hard time to learn fast about Oqtane (in parallel with my other activities) what I would like because I want to rely on Oqtane in my current project.

Here is my longing for help: https://github.com/oqtane/oqtane.docs/issues/4

**Reply by sbwalker** [reply]
When I refer to native state management I am referring to passing information between components. Oqtane utilizes cascading parameters ( ie. PageState, ModuleState ), scoped services ( ie. SiteState ), as well as passing information via the url as embedded segments or querystrings. There are examples throughout the Oqtane code base which demonstrate how to do these things.

**Reply by sbwalker** [reply]
As far as documentation is concerned it would be great if someone from the open source community offered to create some documentation.
